src/qibolab/platforms/platform.py

```diff
@@ -30,14 +36,80 @@ def disconnect(self):
         self.design.disconnect()
         self.is_connected = False
 
-    def execute_pulse_sequence(self, sequence, nshots=1024, relaxation_time=None):
-        if relaxation_time is None:
-            relaxation_time = self.relaxation_time
-        return self.design.play(self.qubits, sequence, nshots=nshots, relaxation_time=relaxation_time)
+    def execute_pulse_sequence(self, sequence, **kwargs):
+        options = ExecutionParameters(**kwargs)
+
+        if options.relaxation_time is None:
+            options.relaxation_time = self.relaxation_time
+        if options.fast_reset:
+            options.fast_reset = {
+                qubit.name: self.create_RX_pulse(qubit=qubit.name, start=sequence.finish)
+                for qubit in self.qubits.values()
+                if not qubit.flux_coupler
+            }
+
+        return self.design.play(
+            self.qubits,
+            sequence,
+            options=options,
+        )
+
+    def sweep(self, sequence, *sweepers, **kwargs):
+        options = ExecutionParameters(**kwargs)
+
+        if options.relaxation_time is None:
+            options.relaxation_time = self.relaxation_time
+        if options.fast_reset:
+            options.fast_reset = {
+                qubit.name: self.create_RX_pulse(qubit=qubit.name, start=sequence.finish)
+                for qubit in self.qubits.values()
+                if not qubit.flux_coupler
+            }
 
-    def sweep(self, sequence, *sweepers, nshots=1024, relaxation_time=None, average=True):
-        if relaxation_time is None:
-            relaxation_time = self.relaxation_time
         return self.design.sweep(
-            self.qubits, sequence, *sweepers, nshots=nshots, relaxation_time=relaxation_time, average=average
+            self.qubits,
+            sequence,
+            *sweepers,
+            options=options,
         )
+
+
+class AcquisitionType(Enum):
+    """
+    Types of data acquisition from hardware.
+    (SPECTROSCOPY[Weird Zurich mode], INTEGRATION, RAW, DISCRIMINATION)
+    """
+
+    RAW = auto()
+    INTEGRATION = auto()
+    DISCRIMINATION = auto()
+
+
+class AveragingMode(Enum):
+    """
+    Types of data averaging from hardware.
+    (CYLIC[True averaging], SINGLESHOT[False averaging], [SEQUENTIAL, bad averaging])
+    """
+
+    CYCLIC = auto()
+    SINGLESHOT = auto()
+
+
+@dataclass
+class ExecutionParameters:
+    """Data structure to deal with execution parameters
+
+    :nshots: Number of shots per point on the experiment
+    :relaxation_time: Relaxation time for the qubit
+    :fast_reset: Enable or disable fast reset
+    :sim_time: Time for the simulation execution
+    :acquisition_type: Data acquisition mode
+    :averaging_mode: Data averaging mode
+    """
+
+    nshots: Optional[np.uint32] = None
+    relaxation_time: Optional[np.uint32] = None
+    fast_reset: Optional[bool] = False
+    sim_time: Optional[np.float64] = 10e-6
```

Comment:
Yup, for now I was trying to do hardware and the simulation right after but it messed everything as it would require to reconnect again on multiple experiment executions, which takes enough time for me to disable simulation and leave it as a debug only tool for now.